app/src/main/java/com/testtask/apptesttask/model/storage/Prefs.kt

```diff
@@ -2,31 +2,35 @@ package com.testtask.apptesttask.model.storage
 
 import android.content.Context
 import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import com.testtask.apptesttask.entity.charactrers.Character
 import javax.inject.Inject
 
-class Prefs @Inject constructor(context: Context, val gson: Gson) {
+class Prefs @Inject constructor(
+    context: Context,
+    private val gson: Gson
+) {
 
     private val PREF_TAG = "characters"
-    private val FAVORITE_CHARACTER = "favorite_characters"
+    private val FAVORITE_CHARACTERS = "favorite_characters"
 
     private val sharedPreferences =
             context.getSharedPreferences(PREF_TAG, Context.MODE_PRIVATE)
 
-    var favoritesCharacters: MutableList<Boolean>?
-        get() = convertArrayToMutableList(sharedPreferences.getString(FAVORITE_CHARACTER, null))
+    var favoritesCharacters: MutableMap<Int, Character>?
+        get() = convertArrayToMutableList(sharedPreferences.getString(FAVORITE_CHARACTERS, null))
         set(value) {
-            sharedPreferences.edit().putString(FAVORITE_CHARACTER, gson.toJson(value))
+            sharedPreferences.edit().putString(FAVORITE_CHARACTERS, gson.toJson(value))
                     .apply()
         }
 
-    private fun convertArrayToMutableList(valueString: String?): MutableList<Boolean> {
-        val mutableList = mutableListOf<Boolean>()
+    private fun convertArrayToMutableList(valueString: String?): MutableMap<Int, Character> {
```

Comment:
Remove this logic and keep `Map` as `JSON` -> https://stackoverflow.com/questions/8360836/gson-is-there-an-easier-way-to-serialize-a-map.